### PR TITLE
[Docs] Update backtracing documentation for Linux.

### PR DESCRIPTION
**Pull request by al45tair**
Update the backtracing documentation with details on static linking, frame pointer usage and a couple of other matters.

rdar://117679284

### CODE REVIEW

**Comment by al45tair** [comment]
@swift-ci Please smoke test

File: docs/Backtracing.rst [hunk 1]
```diff
@@ -229,3 +259,57 @@ Other Darwin (iOS, tvOS)
...
+For users who statically link their binaries and do not wish to ship the Swift
+runtime library alongside them, there is a statically linked copy of
+``swift-backtrace``, named ``swift-backtrace-static`` , in the ``libexec``
+directory alongside the normal ``swift-backtrace`` binary.
+
+By default, to locate ``swift-backtrace``, the runtime will attempt to look in
+the following locations::
+
+    <swift-root>/libexec/swift/<platform>
+    <swift-root>/libexec/swift/<platform>/<arch>
```
Comment (mikeash) [inline docs/Backtracing.rst:290]: Will people know the exact form that `<platform>` and `<arch>` will have? If not, would be good to briefly describe that.

**Review by mikeash — APPROVED** [review]

**Comment by al45tair** [comment]
@swift-ci Please smoke test and merge.

File: docs/Backtracing.rst (hunk 1 above)
Comment (al45tair) [reply docs/Backtracing.rst:290]: I've added a description.

**Pull request merged**